src/audio/dai-zephyr.c

```diff
@@ -1054,7 +1054,12 @@ static int dai_comp_trigger_internal(struct comp_dev *dev, int cmd)
 		dai_trigger_op(dd->dai, cmd, dev->direction);
 #else
 		dai_trigger_op(dd->dai, cmd, dev->direction);
-		ret = dma_stop(dd->chan->dma->z_dev, dd->chan->index);
+		if (dev->state == COMP_STATE_ACTIVE) {
+			ret = dma_stop(dd->chan->dma->z_dev, dd->chan->index);
+		} else {
+			comp_warn(dev, "dma was stopped earlier");
```

Comment:
@tmleman may I ask why we end up stopping the same DMA channel more than once anyway?

Reply:
For example in case of dai-zephyr, when we set PPL to pause it triggers COMP_TRIGGER_PAUSE. This leads to the function dma_suspend being called. Which is really just a call to function intel_adsp_hda_dma_stop.

```c
	.start = intel_adsp_hda_dma_start,
	.stop = intel_adsp_hda_dma_stop,
	.suspend = intel_adsp_hda_dma_stop,
	.get_status = intel_adsp_hda_dma_status,
```

Later when we do PPL reset we are triggering COMP_TRIGGER_STOP and that calls dma_stop again.

Reply:
Thanks for the clarification @tmleman